actions/post_actions.jsx

```diff
@@ -61,33 +61,34 @@ function completePostReceive(post, websocketMessageProps) {
 
 function dispatchPostActions(post, websocketMessageProps) {
     const {currentChannelId} = getState().entities.channels;
+    const channelPostsStatus = getState().views.channel.channelPostsStatus;
 
-    if (post.channel_id === currentChannelId) {
+    if (post.channel_id === currentChannelId && channelPostsStatus[currentChannelId] && channelPostsStatus[currentChannelId].atEnd) {
         dispatch({
             type: ActionTypes.INCREASE_POST_VISIBILITY,
             data: post.channel_id,
             amount: 1,
         });
-    }
 
-    // Need manual dispatch to remove pending post
-    dispatch({
-        type: PostTypes.RECEIVED_POSTS,
-        data: {
-            order: [],
-            posts: {
-                [post.id]: post,
+        // Need manual dispatch to remove pending post
+        dispatch({
+            type: PostTypes.RECEIVED_POSTS,
```

Comment:
Would it be the same if we batch these actions: `INCREASE_POST_VISIBILITY` and `RECEIVED_POSTS`?

Reply:
I am gonna change this with the ticket https://mattermost.atlassian.net/browse/MM-11507. This might not be grouped as it was before But for now, i will batch it so we can revert it later.